x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.test.tsx

```diff
@@ -0,0 +1,167 @@
+const handleAuthProviderChange = jest.fn();
+
+const baseProps = {
+  attributeName: 'An Attribute',
+  attributeValue: 'Something',
+  attributes: ['a', 'b', 'c'],
+  availableAuthProviders: ['ees_saml', 'kbn_saml'],
+  selectedAuthProviders: ['ees_saml'],
+  elasticsearchRoles: ['whatever'],
+  multipleAuthProvidersConfig: true,
+  disabled: false,
+  handleAttributeSelectorChange,
+  handleAttributeValueChange,
+  handleAuthProviderChange,
+};
+
+describe('AttributeSelector', () => {
+  it('renders', () => {
+    const wrapper = shallow(<AttributeSelector {...baseProps} />);
+
+    expect(wrapper.find('[data-test-subj="attributeSelector"]')).toBeDefined();
+  });
+
+  it('renders disabled panel with className', () => {
+    const wrapper = shallow(<AttributeSelector {...baseProps} disabled />);
+
+    expect(wrapper.find('[data-test-subj="attributeSelector"]').prop('className')).toEqual(
+      'euiPanel--disabled'
+    );
+  });
+
+  describe('Auth Providers', () => {
+    const findAuthProvidersSelect = (wrapper: ShallowWrapper) =>
+      wrapper.find('[data-test-subj="authProviderSelect"]');
+
+    it('will not render if "availableAuthProviders" prop has not been provided', () => {
+      const wrapper = shallow(
+        <AttributeSelector {...baseProps} availableAuthProviders={undefined} />
+      );
+
+      expect(wrapper.find('[data-test-subj="authProviderSelect"]')).toHaveLength(0);
+    });
+
+    it('handles fallback props', () => {
+      const wrapper = shallow(
+        <AttributeSelector
+          {...baseProps}
+          attributeValue={undefined}
+          selectedAuthProviders={undefined}
+        />
+      );
+
+      const select = findAuthProvidersSelect(wrapper) as any;
+
+      expect(select.prop('selectedOptions')).toEqual([
+        {
+          label: ANY_AUTH_PROVIDER_OPTION_LABEL,
+          value: ANY_AUTH_PROVIDER,
+        },
+      ]);
+    });
+
+    it('renders a list of auth providers from the "availableAuthProviders" prop including an "Any" option', () => {
+      const wrapper = shallow(
+        <AttributeSelector {...baseProps} availableAuthProviders={['ees_saml', 'kbn_saml']} />
+      );
+      const select = findAuthProvidersSelect(wrapper) as any;
+
+      expect(select.props().options).toEqual([
+        {
+          label: expect.any(String),
+          options: [{ label: 'Any', value: '*' }],
+        },
+        {
+          label: expect.any(String),
+          options: [
+            { label: 'ees_saml', value: 'ees_saml' },
+            { label: 'kbn_saml', value: 'kbn_saml' },
+          ],
+        },
+      ]);
+    });
+
+    it('the "selectedAuthProviders" prop should be used as the selected value', () => {
+      const wrapper = shallow(
+        <AttributeSelector
+          {...baseProps}
+          availableAuthProviders={['ees_saml', 'kbn_saml']}
+          selectedAuthProviders={['kbn_saml']}
+        />
+      );
+      const select = findAuthProvidersSelect(wrapper) as any;
```

Comment:
```suggestion
      const select: ShallowWrapper = findAuthProvidersSelect(wrapper);
```

Constance said there are metrics within Kibana on `any` usage, so we should avoid it when possible, even in tests.

Reply:
You actually may not even need to provide a type here.